Marlin/src/module/configuration_store.cpp

```diff
@@ -397,7 +397,9 @@ void MarlinSettings::postprocess() {
     #define EEPROM_SKIP(VAR) (eeprom_index += sizeof(VAR))
   #endif
 
-  #define EEPROM_START()          int eeprom_index = EEPROM_OFFSET; persistentStore.access_start()
+  int eeprom_index = EEPROM_OFFSET;
+
+  #define EEPROM_START()          persistentStore.access_start()
```

Comment:
This will break EEPROM read/write when done more than once between reboots. `EEPROM_START` must set the `eeprom_index` value back to the start value, otherwise nothing else will set it. The `eeprom_index` value is meant to be a local, and should not be changed into a global.